src/App.tsx

```diff
@@ -1,26 +1,41 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useEffect } from 'react';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import * as todosActions from './features/todos';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const dispatch = useAppDispatch();
+  const { todos, loaded, selectedTodo } = useAppSelector(state => state.todos);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    dispatch(todosActions.init());
+  }, [dispatch]);
+
+  return (
+    <>
+      <div className="section">
+        <div className="container">
+          <div className="box">
+            <h1 className="title">Todos:</h1>
+
+            <div className="block">
+              <TodoFilter />
+            </div>
 
-          <div className="block">
-            <Loader />
-            <TodoList />
+            <div className="block">
+              {loaded ? (
+                <Loader />
+              ) : (
+                <TodoList todos={todos} selectedTodo={selectedTodo} />
+              )}
```

Comment:
The conditional rendering logic here seems inverted. The `Loader` component should be displayed when `loaded` is `false`, indicating that data is still being fetched. Conversely, the `TodoList` should be displayed when `loaded` is `true`, meaning the data has been successfully loaded. Please adjust the condition accordingly.